test/src/queryhandler.spec.ts

```diff
@@ -351,4 +352,34 @@ describe('Query handler tests', function () {
       });
     });
   });
+
+  describe('P selectors', function () {
```

Comment:
let's add tests with multiple query types including custom query handlers. Also, let's add coverage for the waitForSelector user case.

Maybe let's call it Puppeteer selectors in free-form texts?

Reply:
> Also, let's add coverage for the waitForSelector user case.

This is already covered since P queries will be the default,

> Maybe let's call it Puppeteer selectors in free-form texts?

I'm not sure. Puppeteer selectors are already an overloaded term.

We only need to add coverage for each of the query types and custom query handlers.